sympy/solvers/solvers.py

```diff
@@ -1059,6 +1059,15 @@ def _has_piecewise(e):
         if _has_piecewise(fi):
             f[i] = piecewise_fold(fi)
 
+    # Case of `Indexed` objects:
```

Comment:
Since Indexed isn't a Symbol, this is already handled by solve around line 950 (look for `symbol_swapped`. Using sympy.llive.org I get

```
>>> q=Indexed('i',x)
>>> q.is_Symbol
False
>>> solve(q-1,q)
[1]
```

So rather than fix only the Indexed issue, you can fix solveset so it recognizes non-symbol entities like solve does.

Reply:
I don't get the same result with the latest master branch. You're right about `solveset`, but it doesn't look like `solve` yields that result.